src/vs/workbench/contrib/files/browser/fileActions.ts

```diff
@@ -540,7 +540,44 @@ export class ReopenResourcesAction extends Action {
 
 		const options = activeEditorPane.options;
 		const group = activeEditorPane.group;
-		return openEditorWith(activeInput, undefined, options, group, this.editorService, this.configurationService, this.quickInputService);
+		await openEditorWith(activeInput, undefined, options, group, this.editorService, this.configurationService, this.quickInputService);
+	}
+}
+
+export class ToggleEditorTypeCommand extends Action {
+
+	static readonly ID = 'workbench.files.action.toggleEditorType';
```

Comment:
Suggestions welcome for the command name and label

Reply:
To me "toggle" implies that it's two editors (maybe language knowledge gap) so I wonder if this needs to express better what happens when you 2+ editors for a resource?

Reply:
This command does actually behave that way: it switches back and forth between the default and the first available custom editor